### DISCUSSION TITLE
Changing background color by page.draw_rect() when existing white rect

### DISCUSSION THREAD

**Discussion by QQLazy** [pymupdf/PyMuPDF#2067] [Looking for help] [answered]
Hello,
I want to change the PDF's background color from blank to gray, and I used the method mentioned in PyMuPDF's document：

> An efficient way to background-color a PDF page with the old Python paper color is,
>col = fitz.utils.getColor("py_color")
>page.draw_rect(page.rect, color=col, fill=col, overlay=False)

But I find some pdf  have  'white rectangle', after running, the white rectangle still here.Below it's my code, and 
the pdf files for testing is attached.

`doc = fitz.open('test.pdf')`
`for page in doc:`
    `page.draw_rect(page.rect, color=fitz.utils.getColor('lightgray'), fill=fitz.utils.getColor('lightgray'), overlay=False, fill_opacity=0.3)`
`doc.save('test_change.pdf')`

However, some pdf softwares can change the background color perfectly which ignore the white rectangle, like this example.
![截图_选择区域_20221120223055](https://user-images.githubusercontent.com/63276181/202908184-a8559da8-f56d-4d9d-9088-9fe3bdf44ff9.png)

So I wonder if I can use pymupdf to achieve the same effect, the problem bothered me for several days.
Appreciating for your help, thanks.

[test_change.pdf](https://github.com/pymupdf/PyMuPDF/files/10049856/test_change.pdf)
[test.pdf](https://github.com/pymupdf/PyMuPDF/files/10049858/test.pdf)

**Comment by JorjMcKie** [comment, marked as answer]
`overlay=False` cannot work in this case, because anything "coming later" will of course be "above" it - including white rectangles etc.

**Reply by QQLazy** [reply]
Thanks for reply, and I know the overlay parameter controls the drawing or text in foreground or background.
But I wonder if there is a way that I can do to deal with the white rect(since I can detect the rect from get_drawings()).
I have thought about two methods:
1. Change the fill color of the rect drawings which is white;
2. Drawing a rect below the text and above the existing rect.

Are the methods I mentioned above can be  useful？Or there's another way I can follow.🙏

**Reply by JorjMcKie** [reply]
No, this is impossible.

**Reply by QQLazy** [reply]
All right, then I'll let it go, thank you for your help anyway.👌

**Reply by QQLazy** [reply]
I find that I have made a mistake, the software I thought is useful didn't do anything special, it puts layers in front by default.

So the following code is just the answer I want, if only you let the `fill_opacity` be small and let `overlay=True`.

> page.draw_rect(page.rect, color=col, fill=col, overlay=False)
